src/Appwrite/Auth/Validator/PasswordHistory.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Appwrite\Auth\Validator;
+
+use Appwrite\Auth\Auth;
+
+/**
+ * Password.
+ *
+ * Validates user password string
+ */
+class PasswordHistory extends Password
+{
+    protected array $history;
+
+    public function __construct(array $history, string $algo, array $algoOptions = [])
+    {
+        $this->history = $history;
+        $this->algo = $algo;
+        $this->algoOptions = $algoOptions;
+    }
+
+    /**
+     * Get Description.
+     *
+     * Returns validator description
+     *
+     * @return string
+     */
+    public function getDescription(): string
+    {
+        return 'Password shouldn\'t be in the history.';
+    }
+
+    /**
+     * Is valid.
+     *
+     * @param mixed $value
+     *
+     * @return bool
+     */
+    public function isValid($value): bool
+    {
+        foreach ($this->history as $hash) {
```

Comment:
Why do we validate the password here? Don't we just need to check if the new hash in the history list of hashes?

Reply:
tried using `in_array` didn't work. I had to use `Auth::passwordVerify` not sure why.